File: lib/data/models/auth/login_request.dart
class LoginRequest {
  final String identifier;
  final String password;
  // final String? schoolKey;

  LoginRequest({
    required this.identifier,
    required this.password,
    // this.schoolKey,
  });

  Map<String, dynamic> toJson() {
    return {
      'identifier': identifier,
      'password': password,
      // if (schoolKey != null) 'school_key': schoolKey,
    };
  }
}

File: app/Http/Requests/Student/V01/Auth/LoginRequest.php
<?php

namespace App\Http\Requests\Student\V01\Auth;

use Illuminate\Foundation\Http\FormRequest;

class LoginRequest extends FormRequest
{
    /**
     * Get the validation rules that apply to the request.
     *
     * @return array<string, mixed>
     */
    public function rules()
    {
        return [
            'login' => 'nullable|string|required_without_all:email,phone',
            'phone' => 'nullable|string|required_without_all:login,email',
            'email' => 'nullable|string|required_without_all:login,phone',
            'password' => 'required|string|min:6',
            'confirm' => 'nullable|boolean',
            // 'school_key' => 'required|string',
        ];
    }
}
